Civi/Core/Service/AutoDefinition.php

```diff
@@ -124,12 +123,6 @@ protected static function applyTags(Definition $def, \ReflectionClass $class, ar
     if ($class->implementsInterface(SpecProviderInterface::class)) {
       $def->addTag('spec_provider');
     }
-
-    if (!empty($classDoc['serviceTags'])) {
-      foreach (static::splitSymbols($classDoc['serviceTags']) as $extraTag) {
-        $def->addTag($extraTag);
-      }
-    }
```

Comment:
@totten this code never worked because the variable `$classDoc` is undefined.
I think it's safe to remove as it smells like YAGNI.

Reply:
In this context, maybe should be `$docBlock` instead of `$classDoc`.

Re: YAGNI: The thing is that - right now - Civi does use *some* service-tags, but it does it in a weird way -- where they're like ballerina-ghosts. (They're invisible, and they dance around in circles.) I don't think Symfony-upstream conceived them as ballerina-ghosts. The question is whether one thinks the general notion of "service tags" should be enlivened in Civi.

* If no, then... `@serviceTags` will never have a use-case.
* If yes, then... `@serviceTags` will be a better model than the other tag-related stuff (i.e. more transparent, less glue code).

My fear is this: today, we say: "Nah, don't need service tags." Yay, simplicity. Then 6 months from now, someone sees the pirouette-tag stuff and adds more pirouettes (because there's nothing simpler available).

Reply:
I'm not quite sure I can picture what a ballerina ghost looks like in PHP code! But my general approach when I invent a better coding pattern is to bulk update all the older patterns in core. Because if I don't then people will continue to follow those examples.

Reply:
OK, so we chatted a bit by phone.

One idea that Coleman raised was to deal with the issue of future interfaces/tags by allowing the declaration on the class. I was in denial about needing work, so I just wanted to put a comment to encourage that... but as I got to writing, I figured that writing the description was half the job, so I might as well do the other half. And so here's the PR with description:

https://github.com/civicrm/civicrm-core/pull/26963